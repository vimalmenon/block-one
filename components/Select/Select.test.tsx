import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Select } from './'

describe('Select Testing', () => {
  test('Select: Snapshot testing', () => {
    const onChange = jest.fn()
    const items = [
      {
        name: 'one',
        value: 'one',
      },
      {
        name: 'two',
        value: 'two',
      },
    ]

    const wrapper = shallow(<Select items={items} nameProps="name" onChange={onChange} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  test('Select: Snapshot testing without nameProps', () => {
    const onChange = jest.fn()
    const items = [
      {
        name: 'one',
        value: 'one',
      },
      {
        name: 'two',
        value: 'two',
      },
    ]

    const wrapper = shallow(<Select items={items} onChange={onChange} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  test('Select: Testing Change Event', () => {
    const onChange = jest.fn()
    const items = [
      {
        name: 'one',
        value: 'one',
      },
      {
        name: 'two',
        value: 'two',
      },
    ]

    const wrapper = shallow(<Select items={items} nameProps="name" onChange={onChange} />)
    wrapper.find('select').simulate('change')
    expect(onChange).toHaveBeenCalled()
  })
})
