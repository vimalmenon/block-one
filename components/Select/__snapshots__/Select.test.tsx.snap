// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Testing Select: Snapshot testing 1`] = `
<select
  className="py-2 px-4 focus:ring-blue-600 focus:border-blue-500 outline-none block w-full sm:text-sm border border-indigo-100 rounded-md"
  onChange={[MockFunction]}
  value=""
>
  <option
    key="0"
    value="{\\"name\\":\\"one\\",\\"value\\":\\"one\\"}"
  >
    one
  </option>
  <option
    key="1"
    value="{\\"name\\":\\"two\\",\\"value\\":\\"two\\"}"
  >
    two
  </option>
</select>
`;

exports[`Select Testing Select: Snapshot testing without nameProps 1`] = `
<select
  className="py-2 px-4 focus:ring-blue-600 focus:border-blue-500 outline-none block w-full sm:text-sm border border-indigo-100 rounded-md"
  onChange={[MockFunction]}
  value=""
>
  <option
    key="0"
    value="{\\"name\\":\\"one\\",\\"value\\":\\"one\\"}"
  >
    one
  </option>
  <option
    key="1"
    value="{\\"name\\":\\"two\\",\\"value\\":\\"two\\"}"
  >
    two
  </option>
</select>
`;
